src/config/mongoose_config_spec.erl

```diff
@@ -282,17 +282,18 @@ xmpp_listener_common() ->
     #section{items = #{<<"backlog">> => #option{type = integer,
                                                 validate = non_negative},
                        <<"proxy_protocol">> => #option{type = boolean},
-                       <<"hibernate_after">> => #option{type = integer,
+                       <<"hibernate_after">> => #option{type = int_or_infinity,
                                                         validate = non_negative},
                        <<"max_stanza_size">> => #option{type = int_or_infinity,
-                                                        validate = positive},
+                                                        validate = positive,
+                                                        process = fun process_infinity_as_zero/1},
                        <<"num_acceptors">> => #option{type = integer,
-                                               validate = positive}
+                                                      validate = positive}
                       },
-             defaults = #{<<"backlog">> => 100,
+             defaults = #{<<"backlog">> => 1024,
                           <<"proxy_protocol">> => false,
                           <<"hibernate_after">> => 0,
-                          <<"max_stanza_size">> => infinity,
```

Comment:
Why change it? I think it makes more sense than after the change, and we use the atom `infinity` quite consistently throughout the code. OTP also uses it.

Reply:
This one because the `process` function is ran _after_ the defaults, not before, so if I leave infinity as default, then c2s will get an infinity and will have to do the conversion for `exml`.

Reply:
I meant "why not keep `infinity`". I see taht `exml` uses zero. Let's keep it.